src/playback-watcher.js

```diff
@@ -205,18 +218,23 @@ export default class PlaybackWatcher {
     this.resetSegmentDownloads_(type);
     this.tech_.trigger({type: 'usage', name: `vhs-${type}-download-exclusion`});
 
-    if (type !== 'subtitle') {
-      mpc.blacklistCurrentPlaylist({
-        message: `Infinite ${type} segment downloading detected.`
-      }, Infinity);
-    } else {
+    if (type === 'subtitle') {
+      // TODO: Is there anything else that we can do here?
+      // removing the track and disabling could have accesiblity implications.
       const track = loader.track();
       const label = track.label || track.language || 'Unknown';
 
       videojs.log.warn(`Text track "${label}" is not working correctly. It will be disabled and excluded.`);
       track.mode = 'disabled';
       this.tech_.textTracks().removeTrack(track);
+      return;
     }
+
+    // TODO: should we exclude audio tracks rather than main tracks
+    // when type is audio?
+    mpc.blacklistCurrentPlaylist({
+      message: `Infinite ${type} segment downloading detected.`
```

Comment:
A bit minor, but since this can potentially be seen by users, I think we'd be better off phrasing it along the lines of "Excessive" instead of "Infinite," as it more accurately reflects what's detected.